### PR TITLE
maint: dockerizing flask app

### LINKED ISSUE #72: dockerize the example flask app
We have a dockerized version of the plain python app, but this ticket is to dockerize the flask app to test more instrumentation and baggage with smoke tests

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- Closes #72 

## Short description of the changes

- add Dockerfile for flask app
- update `opentelemetry-api`, `opentelemetry-sdk`, and `opentelemetry-exporter-otlp` from `1.14.0` to `1.16.0` to help deal with version conflicts
- update `opentelemetry-instrumentation` and `opentelemetry-instrumentation-flask` from `0.36b0` to `0.37b0` to help deal with version conflicts
- update `pylint`
- add flask app to docker-compose
- whitespace cleanup

## How to verify that this has the expected result

Build the flask app, and locally `curl localhost:5000`:

`cd smoke-tests && docker-compose up --build app-sdk-grpc-flask`

### CODE REVIEW

**Comment by emilyashley** [comment]
I just made a lil PR of some changes to bring into this branch hoping to sidestep some of the errors (#94 )

**Review by emilyashley — APPROVED** [review]
✨

**Comment by emilyashley** [comment]
Wait! Don't merge this yet. I think my OTel Dependency updates don't work as fully as expected. Let me check on that today before this gets pulled in 🙏

**Comment by emilyashley** [comment]
I just needed to remove my poetry env and start fresh. Ignore this. We're good to go. ✨

**Comment by JamieDanielson** [comment]
NOTE: I merged in another PR for smoke testing the plain Python app; this PR will now need a rebase, and some changes for the smoke tests and docker. We can run as [separate smoke tests like we do in node](https://github.com/honeycombio/honeycomb-opentelemetry-node/blob/main/smoke-tests/docker-compose.yml).

File: examples/hello-world-flask/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+COPY src/ ./src/
+RUN poetry install
+
+# Copy the examples into the image and install deps.
+COPY examples/ ./examples/
+RUN cd ./examples/hello-world-flask && poetry install
+
+# From this point forward, we're operating on the flask example.
+WORKDIR /app/examples/hello-world-flask
+CMD ["poetry", "run", "opentelemetry-instrument", "flask", "run", "--host", "0.0.0.0"]
```
Comment (JamieDanielson) [inline examples/hello-world-flask/Dockerfile:31]: Apparently we need to pass the host in the command line argument even though it's specified in the app... many thanks to [ this StackOverflow ](https://stackoverflow.com/questions/30323224/deploying-a-minimal-flask-app-in-docker-server-connection-issues/58138250)for guidance

File: pyproject.toml [hunk 2]
```diff
@@ -8,10 +8,10 @@ packages = [{include = "honeycomb", from = "src" }]
 
 [tool.poetry.dependencies]
 python = "^3.7, >= 3.7.2"
-opentelemetry-api = "^1.14.0"
-opentelemetry-sdk = "^1.14.0"
-opentelemetry-exporter-otlp = "^1.14.0"
-opentelemetry-instrumentation = "^0.36b0"
+opentelemetry-api = "^1.16.0"
```
Comment (JamieDanielson) [inline pyproject.toml:11]: All these dependency updates helped us get over version conflicts that kept cropping up in this work... but I wonder if we should merge those in separately as a standalone PR, then rebase this to pull those changes in?

**Review by MikeGoldsmith — APPROVED** [review]

**Pull request merged**